tools/src/main/java/org/apache/kafka/trogdor/workload/ConsumeBenchWorker.java

```diff
@@ -88,41 +99,78 @@ public void start(Platform platform, WorkerStatusTracker status,
         @Override
         public void run() {
             try {
-                executor.submit(consumeTask());
+                List<Future<Void>> consumeTasks = new ArrayList<>();
+                for (ConsumeMessages task : consumeTasks()) {
+                    consumeTasks.add(executor.submit(task));
+                }
+                executor.submit(new CloseStatusUpdater(consumeTasks));
             } catch (Throwable e) {
                 WorkerUtils.abort(log, "Prepare", e, doneFuture);
             }
         }
 
-        private ConsumeMessages consumeTask() {
-            String consumerGroup = spec.consumerGroup();
+        private List<ConsumeMessages> consumeTasks() {
+            List<ConsumeMessages> tasks = new ArrayList<>();
+            String consumerGroup = consumerGroup();
+            int consumerCount = spec.consumerCount();
             Map<String, List<TopicPartition>> partitionsByTopic = spec.materializeTopics();
-            boolean toUseGroupPartitionAssignment = partitionsByTopic.values().isEmpty();
+            boolean toUseGroupPartitionAssignment = partitionsByTopic.values().stream().allMatch(List::isEmpty);
+
+            if (!toUseGroupPartitionAssignment && !toUseRandomConsumeGroup() && consumerCount > 1)
+                throw new ConfigException(String.format("Will not split partitions across consumers from the %s group", consumerGroup));
+
+            String clientId = clientId(0);
+            consumer = consumer(consumerGroup, clientId);
+            if (!toUseGroupPartitionAssignment) {
```

Comment:
Since we're handling both the case where this is true, and where it is false, we might as well test for (toUseGroupPartitionAssignment) rather than (!toUseGroupPartitionAssignment)

Reply:
True, that's more readable